src/loaders/sass.ts

```diff
@@ -1,20 +1,29 @@
+import { dirname, basename } from "pathe";
 import type { Loader, LoaderResult } from "../loader";
 
-export const sassLoader: Loader = async (input, { options }) => {
+export const sassLoader: Loader = async (input) => {
   if (![".sass", ".scss"].includes(input.extension)) {
     return;
   }
-
+  if (basename(input.srcPath).startsWith("_")) {
+    // sass files starting with "_" are always considered partials
+    // and should not be compiled to standalone CSS
+    return [{
+      contents: "",
+      path: input.path,
+      skip: true
+    }];
+  }
   const compileString = await import("sass").then(r => r.compileString || r.default.compileString);
 
   const output: LoaderResult = [];
 
   const contents = await input.getContents();
 
   output.push({
-    contents: compileString(contents).css,
+    contents: compileString(contents, { loadPaths: [dirname(input.srcPath)] }).css,
```

Comment:
`loadPaths` is required to enable relative imports (https://github.com/sass/sass/issues/3366#issuecomment-1203193294)

Reply:
@dwightjack what about adding `dirname(input.srcPath, 'node_modules')` ? Just in case we want to add `@use '@external-node-module/style.css'`

Reply:
@cpreston321 thanks for your feedback. I update the PR including `node_modules` and added a test (https://github.com/unjs/mkdist/pull/109/commits/014112006fdf7c5f21aa36b490439f9a3d77a958).

Reply:
Hey @dwightjack the easiest way to to do real test is to install `@unocss/reset` as devDep and make a scss file that has `@use '@unocss/reset/tailwind.css'` to see it gets imported correctly I am thinking that maybe it needs to resolve the full node_modules path but I could be wrong.